src/main/java/net/minecraftforge/common/brewing/BrewingContainerWrapper.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.brewing;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+
+/**
+ * Wraps a container representing a brewing stand
+ */
+public record BrewingContainerWrapper(ItemStack base, ItemStack reagent) implements Container
```

Comment:
In my mods, I tend to implement container wrappers as wrapping my inventory capability instance, as then I don't have to create a new instance for every recipe call. Making this a record means a lot more objects are generated.

In my opinion, using an interface in the recipe is cleaner, one that extends container, defaults unused methods, and includes the base and reagent. You could still have a record implement the interface for the forge implementation like this, or create an implementation that is specifically wrapping the brewing stand tile entity.

----

If this method is preferred (instance per recipe call), I think the java doc/name is misleading, as this does not wrap a container, it wraps two item stacks. Its really a `BrewingContainerRecord` to reuse the java term, a record acting as a container.

Reply:
1. it has to be like that to be able to brew different recipes with the same reagent at the same time
2. I forgot to change the javadoc when I changed the class

Reply:
No? You can make more than one wrapper for a single TE. One of my mods has what I will summarize as a multi-furnace, I create a wrapper per slot.

In any case, that is an implementation detail, you could use this design just fine, I think its just not a great idea to force other mods implementing brewing stands to this design. An interface that you implement would give them full control.